native/src/main/java/io/ballerina/lib/data/xmldata/xml/QualifiedNameMap.java

```diff
@@ -56,13 +57,26 @@ public boolean contains(QualifiedName qName) {
             return false;
         }
         for (QualifiedName qualifiedName : stringToQNameMap.get(localName)) {
-            if (qualifiedName.getNamespaceURI().equals(Constants.NS_ANNOT_NOT_DEFINED)) {
+            if (isSameNamespace(qualifiedName, qName)
+                    && isSameAttributeFlag(qualifiedName.getAttributeState(), qName.getAttributeState())) {
                 return true;
             }
         }
         return false;
     }
 
+    private boolean isSameNamespace(QualifiedName q1, QualifiedName q2) {
```

Comment:
This function can be import from datautils ryt?

Reply:
Updated https://github.com/ballerina-platform/module-ballerina-data.xmldata/pull/29/commits/efe4da67d8f15b671d88b7b9179f7793c15280fb